connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/DynamicPropertyManager.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.inbound;
+
+import java.util.Map;
+
+/**
+ * Manages dynamic properties associated with a given property ID. This interface provides methods
+ * to set, update, retrieve, and check the readiness of properties for binding, as well as clearing
+ * the properties.
+ */
+public interface DynamicPropertyManager {
+
+  /**
+   * Sets or updates the properties associated with the specified property ID.
```

Comment:
I dont see a need for this. Could be removed as the client only deals with `ProcessInstanceContext`.